src/main/java/org/folio/entlinks/service/links/LinksSuggestionService.java

```diff
@@ -177,16 +179,40 @@ private void actualizeBibSubfields(FieldParsedContent bibField,
       return;
     }
 
-    var zeroValue = sourceFileCodeRepository.findByCodeAsPrefixFor(authority.getNaturalId())
-        .map(AuthoritySourceFileCode::getAuthoritySourceFile)
-        .map(sourceFile -> getSubfield0Value(authority.getNaturalId(), sourceFile))
-        .orElse(authority.getNaturalId());
+    var zeroValue = getSubfieldZeroValue(authority);
+
     bibSubfields.put("0", List.of(zeroValue));
     bibSubfields.put("9", List.of(authority.getId().toString()));
 
     modifySubfields(optionalField.get().getSubfields(), bibSubfields, rule);
   }
 
+  private String getSubfieldZeroValue(AuthorityParsedContent authority) {
+    var sourceFileId = authority.getSourceFileId();
+    if (isNull(sourceFileId)) {
+      return authority.getNaturalId();
+    }
+
+    var authoritySourceFile = authoritySourceFileService.getById(sourceFileId);
+    var naturalIdPrefix = extractNaturalIdPrefix(authority.getNaturalId());
+
+    return authoritySourceFile.getAuthoritySourceFileCodes()
+        .stream()
+        .map(AuthoritySourceFileCode::getCode)
+        .filter(naturalIdPrefix::equals)
+        .findFirst()
+        .map(code -> getSubfield0Value(authority.getNaturalId(), authoritySourceFile))
+        .orElse(authority.getNaturalId());
```

Comment:
We don't need to do prefix comparison. It will match in all possible cases. Because source file was assigned to the authority only when we have code-prefix match.
The one thing there that should be done is:
1. Get source file by id
2. Use it in `getSubfield0Value` method